### PR TITLE
smart patient select page

### PR DESCRIPTION
**Pull request by VaishSiddharth**
Issue
https://issues.openmrs.org/browse/FM2-261

### CODE REVIEW

File: omod/src/main/resources/findPatient.gsp [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+<% if (breadcrumbs) { %>
+    var breadcrumbs = ${ breadcrumbs };
+<% } else { %>
+    var breadcrumbs = [
+        { icon: "icon-home", link: '/' + OPENMRS_CONTEXT_PATH + '/index.htm' },
+        { label: "${ ui.message(label)}"}
+    ];
+<% } %>
+
+    jq(function() {
```
Comment (VaishSiddharth) [inline omod/src/main/resources/findPatient.gsp]: @ibacher I think that we might not need a duplicate this page.

What's your say? Also I am not sure that if I have missed out some configuration

File: omod/src/main/java/org/openmrs/module/smartonfhir/web/controller/FindPatientPageAndAddTokenController.java [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+import org.openmrs.module.appui.UiSessionContext;
+import org.openmrs.module.coreapps.helper.BreadcrumbHelper;
+import org.openmrs.ui.framework.UiUtils;
+import org.openmrs.ui.framework.page.PageModel;
+import org.springframework.web.bind.annotation.RequestParam;
+
+/**
+ *
+ */
+public class FindPatientPageAndAddTokenController {
```
Comment (ibacher) [inline omod/src/main/java/org/openmrs/module/smartonfhir/web/controller/FindPatientPageAndAddTokenController.java]:
```suggestion
public class FindPatientPageAndAddTokenController {
```

File: pom.xml [hunk 3]
```diff
@@ -284,6 +292,20 @@
...
+			<version>${appframeworkVersion}</version>
+			<scope>provided</scope>
+		</dependency>
+
+		<dependency>
+			<groupId>org.openmrs.module</groupId>
+			<artifactId>coreapps-omod</artifactId>
+			<version>1.28.0</version>
+			<scope>provided</scope>
+		</dependency>
```
Comment (ibacher) [inline pom.xml:308]: Do we need these here or can we put these in the dependency management section instead?

Comment (ibacher) [reply pom.xml:308]: Also, do we even actually need these?

File: omod/src/main/resources/findPatient.gsp (hunk 1 above)
Comment (ibacher) [reply omod/src/main/resources/findPatient.gsp]: I *think* we need this to keep the connection between the controller and the page, although, I would note that this should actually we be in `omod/src/main/webapp/pages`, which I think will ensure it's in the right place.

File: omod/src/main/resources/findPatient.gsp [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+        [ afterSelectedUrl: afterSelectedUrl,
+          showLastViewedPatients: showLastViewedPatients,
+          breadcrumbOverride: breadcrumbs,
+          registrationAppLink: registrationAppLink])}
+<% } else { %>
+${ ui.includeFragment("coreapps", "patientsearch/patientSearchWidget",
+        [ afterSelectedUrl: afterSelectedUrl,
+          showLastViewedPatients: showLastViewedPatients,
+          registrationAppLink: registrationAppLink ])}
+<% } %>
```
Comment (ibacher) [inline omod/src/main/resources/findPatient.gsp]:
```suggestion
<% } %>

```

File: omod/src/main/java/org/openmrs/module/smartonfhir/web/controller/FindPatientPageAndAddTokenController.java [hunk 5]
```diff
@@ -0,0 +1,48 @@
+/*
+ * This Source Code Form is subject to the terms of the Mozilla Public License,
+ * v. 2.0. If a copy of the MPL was not distributed with this file, You can
+ * obtain one at http://mozilla.org/MPL/2.0/. OpenMRS is also distributed under
+ * the terms of the Healthcare Disclaimer located at http://openmrs.org/license.
+ *
+ * Copyright (C) OpenMRS Inc. OpenMRS is a registered trademark and the OpenMRS
+ * graphic logo is a trademark of OpenMRS Inc.
+ */
+package org.openmrs.module.smartonfhir.web.controller;
```
Comment (ibacher) [inline omod/src/main/java/org/openmrs/module/smartonfhir/web/controller/FindPatientPageAndAddTokenController.java]: This should be in package `org.openmrs.module.smartonfhir.page.controller`.

**Review by ibacher — CHANGES_REQUESTED** [review]

File: pom.xml (hunk 3 above)
Comment (VaishSiddharth) [reply pom.xml:308]: @ibacher I think we need these because we are using `BreadcrumbHelper` and `AppDescriptor` in the controller class

Comment (ibacher) [reply pom.xml:308]: That makes some kind of sense, but I think in that case:

1. They should go in the dependency management section so only the OMOD depends on them.
2. We should modify the config.xml to ensure that those are noted as required modules for this module.

**Comment by VaishSiddharth** [comment]
@ibacher  Please close this PR. I don't think it's needed now

**Pull request closed without merging**